WordPress/openverse-frontend
# Use a non-versioned API URL to mock analytics requests, too

## Fixes


Fixes #915 by @obulat

## Description


This PR creates two ApiService objects: one is versioned, for most endpoints, and an unversioned one for the analytics endpoints. 
Some data services were also refactored to reuse code from other services (`getMediaSlug`, for example), and the `bug-report` was removed as it is unused.
The proxy port was also changed from `3000` to `3322` because when I was testing, the API was running in Docker, and `3000` port was busy.
It also renames the `update-tapes` to `recreate-tapes`, and adds an `update-tapes` script that does not completely delete all the tapes, but simply adds any tapes that are missing. This way, we can update a single tape when any e2e test changes, and don't have too many changes in a PR.

## Testing Instructions

Enable internal analytics:
https://github.com/WordPress/openverse-frontend/blob/690f52360a6d7389f4015ed384b62d36b29b65e6/src/utils/env.ts#L9
Run `pnpm run update-tapes`. This will delete all the tapes files, and, more importantly, it will save any response that isn't found in `test/tapes` directory. You should see a lot of tapes for analytics responses. They are all 404, but they **_are_** there.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
